compiler/AST/iterator.cpp

```diff
@@ -553,12 +551,10 @@ buildZip3(IteratorInfo* ii, Vec<BaseAST*>& asts, BlockStmt* singleLoop) {
                                         new CallExpr(PRIM_CAST, moreType->symbol, condExpr)));
     zip3body->insertAtTail(new CallExpr(PRIM_SET_MEMBER, ii->zip3->_this,
                                         ii->iclass->getField("more"), condTemp));
-  } else if (ForLoop* forLoop = toForLoop(singleLoop)) {
-    SymExpr* index = forLoop->indexGet()->copy(&map);
-
-    zip3body->insertAtTail(new CondStmt(index,
-                                        new CallExpr(PRIM_SET_MEMBER, ii->zip3->_this, ii->iclass->getField("more"), new_IntSymbol(1)),
-                                        new CallExpr(PRIM_SET_MEMBER, ii->zip3->_this, ii->iclass->getField("more"), new_IntSymbol(0))));
+  } else if (isCForLoop(singleLoop)) {
+    // Do nothing.
+  } else {
+    INT_FATAL("Unexpected singleLoop iterator type");
```

Comment:
I might be tempted to add a comment that notes that ForLoops have all been removed from the AST by expandForLoop() by this point, DoWhile loops aren't handled in isSingleLoopIterator(), and param for loops should already have been expanded and removed.

Not sure if I feel strongly about that or not, I might just note it in the commit message as an explanation for why we know there won't be other loop types and why the INT_FATAL is safe.